File: weather_script.js
const fetch = require('node-fetch'); // Ensure you have node-fetch installed (npm install node-fetch)

async function fetchWeather(cityName, apiKey) {
    const baseUrl = "http://api.openweathermap.org/data/2.5/weather";
    const url = `${baseUrl}?q=${cityName}&appid=${apiKey}&units=metric`; // Use "imperial" for Fahrenheit

    try {
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error(`HTTP error! Status: ${response.status}`);
        }
        const weatherData = await response.json();
        return weatherData;
    } catch (error) {
        return { error: error.message };
    }
}

(async () => {
    const API_KEY = "your_api_key_here"; // Replace with your OpenWeatherMap API key
    const city = "London"; // Replace with the desired city name

    const weatherData = await fetchWeather(city, API_KEY);

    if (weatherData.error) {
        console.log(`Error: ${weatherData.error}`);
    } else {
        console.log(`Weather in ${weatherData.name}:`);
        console.log(`Temperature: ${weatherData.main.temp}°C`);
        console.log(`Weather: ${weatherData.weather[0].description.charAt(0).toUpperCase() + weatherData.weather[0].description.slice(1)}`);
        console.log(`Humidity: ${weatherData.main.humidity}%`);
    }
})();

File: weather_script.py
import requests

def fetch_weather(city_name, api_key):
    """
    Fetch weather data for a given city from OpenWeatherMap API.

    Args:
        city_name (str): Name of the city to fetch weather for.
        api_key (str): Your OpenWeatherMap API key.

    Returns:
        dict: Weather data if successful, or an error message.
    """
    base_url = "http://api.openweathermap.org/data/2.5/weather"
    params = {
        "q": city_name,
        "appid": api_key,
        "units": "metric"  # Use "imperial" for Fahrenheit
    }

    try:
        response = requests.get(base_url, params=params)
        response.raise_for_status()  # Raise HTTPError for bad responses (4xx and 5xx)
        return response.json()
    except requests.exceptions.RequestException as e:
        return {"error": str(e)}

if __name__ == "__main__":
    # Replace with your OpenWeatherMap API key
    API_KEY = "your_api_key_here"
    city = input("Enter the city name: ")

    weather_data = fetch_weather(city, API_KEY)

    if "error" in weather_data:
        print(f"Error: {weather_data['error']}")
    else:
        print(f"Weather in {weather_data['name']}:")
        print(f"Temperature: {weather_data['main']['temp']}°C")
        print(f"Weather: {weather_data['weather'][0]['description'].capitalize()}")
        print(f"Humidity: {weather_data['main']['humidity']}%")